src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -91,7 +93,9 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
             logging.info("About to call insert block for: %s", name)
             dbsApi.insertBulkBlock(blockDump=block)
             results.put({'name': name, 'success': "uploaded"})
-        except Exception as ex:
+        except HTTPError as ex:
+            dbsError = DBSError(ex.body)
+            results.put({'name': name, 'success': 'error', 'error': dbsError.Reason()})
```

Comment:
This line has to go away, because you still don't know if this request is supposed to be marked as failed or not (and the dbsError.Reason would crash here as well).

Reply:
Alan, I think the entire block here is obsolete since it relies on DBS python exception. This is why I called DBSError here to get DBSError object. If it fails to parse exception coming from DBS server it will still has original exception as string and `getReason` will return it.

What I think should be done here is rather different, something like this:
```
dbsError = DBSError(ex.body)
dbsCode = dbsError.getServerCode()
if dbsCode == XXX:
   # it is missing block
else if dbsCode == YYY:
   # it is something else
....
```
but until we define these low level errors (which I can do now if we decide on them), then we should still parse exception (or DBS Go `"error":{"reason":...}`.

Please review again to make it clear how to adjust this codebase.